### PR TITLE
Properly skip printing version header with `--no-header`

### PR DESCRIPTION
**Pull request by enetheru**
Check engine::_print_header before actually printing header info to the console.

### CODE REVIEW

**Comment by AThousandShips** [comment]
As I suggested on the other PR I think this should be handled in `Engine` not here

**Comment by akien-mga** [comment]
Updated with @AThousandShips' suggested improvement.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
For the record, this was implemented properly in #79179, but it was broken when rebasing #59247.

**Comment by AThousandShips** [comment]
I was wondering why it wasn't used, that explains it

File: main/main.cpp [hunk 1]
```diff
@@ -395,18 +395,21 @@ void finalize_theme_db() {
 #endif
 
 void Main::print_header(bool p_rich) {
+	if (!Engine::get_singleton()->_print_header) {
+		return;
+	}
```
Comment (AThousandShips) [inline main/main.cpp:400]:
```suggestion
```
Redundant

**Review by AThousandShips — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**